### PR TITLE
[SW-409] qr code reader

### PR DESCRIPTION
**Pull request by ifaouibadi**

### CODE REVIEW

File: src/popup/components/Modals/QrCodeReader.vue [hunk 1]
```diff
@@ -95,8 +96,10 @@ export default {
                 );
                 reject();
               }
-              if (navigator.mediaDevices.getUserMedia) {
+              if (navigator.mediaDevices && navigator.mediaDevices.getUserMedia) {
```
Comment (konradodo) [inline src/popup/components/Modals/QrCodeReader.vue:99]: navigator.mediaDevices?.getUserMedia

File: src/popup/components/Modals/QrCodeReader.vue [hunk 2]
```diff
@@ -112,6 +115,10 @@ export default {
     },
   },
   async mounted() {
+    console.info('========================');
```
Comment (konradodo) [inline src/popup/components/Modals/QrCodeReader.vue:118]: remove console log

File: src/popup/components/Modals/QrCodeReader.vue [hunk 3]
```diff
@@ -166,9 +189,17 @@ export default {
...
           }, 500);
         })
-        : (
-          await this.browserReader.decodeFromInputVideoDevice(undefined, this.$refs.qrCodeVideo)
-        ).getText();
+        : new Promise((resolve, reject) => {
+          this.browserReader.decodeFromVideoDevice(
+            this.videoInputDevices[0].deviceId,
+            this.webQrControls = this.$refs.qrCodeVideo, (result, errors, controls) => {
+              if (result && result.text) {
```
Comment (konradodo) [inline src/popup/components/Modals/QrCodeReader.vue:196]: result?.text

File: src/popup/components/Modals/QrCodeReader.vue [hunk 4]
```diff
@@ -179,7 +210,13 @@ export default {
         window.plugins.webviewcolor.change('#141414');
         this.setQrScanner(false);
         window.QRScanner.destroy();
-      } else this.browserReader.reset();
+      } else {
+        try {
```
Comment (konradodo) [inline src/popup/components/Modals/QrCodeReader.vue:214]: we should do somethink with catch, or.. if is not nessesary remove the try catch

@peronczyk @pstachel what do you think?

File: src/popup/components/QrCode.vue [hunk 5]
```diff
@@ -31,15 +31,15 @@ export default defineComponent({
...
-      imageOptions: {
-        hideBackgroundDots: false,
-        imageSize: 0.4,
-        margin: 0,
-      },
-      dotsOptions: { type: 'square', color: '#1161fe' },
-      image: SHLogo,
-      cornersSquareOptions: { type: 'square', color: '#6da0fe' },
-      cornersDotOptions: { type: 'square', color: '#1161fe' },
+      // imageOptions: {
```
Comment (konradodo) [inline src/popup/components/QrCode.vue:34]: pls remove the comment

**Review by pstachel — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -51,6 +51,7 @@
     "detect-browser": "^5.3.0",
     "dex-contracts-v2": "github:aeternity/dex-contracts-v2",
     "lodash-es": "^4.17.21",
+    "qr-code-styling": "github:sallandpioneers/qr-code-styling",
```
Comment (peronczyk) [inline package.json:54]: In my opinion we should not add any external packages not handled by the NPM as there is a risk of someone taking over the repo.

**Review by peronczyk — CHANGES_REQUESTED** [review]

**Pull request merged**